toolbox/repo/validate_role_files.py

```diff
@@ -0,0 +1,68 @@
+#! /usr/bin/env python
+
+import glob, os
+import yaml
+import pathlib
+
+THIS_DIR = pathlib.Path(__file__).absolute().parent
+
+# go to top directory
+os.chdir(THIS_DIR.parent.parent)
+
+ROLES_VARS_GLOB = "roles/*/vars/*/*"
+FILE_PREFIX = "roles/"
+
+def validate_role_vars_files(filename, yaml_doc):
+    errors = 0
+
+    for key, value in yaml_doc.items():
+        try:
+
+            if not value.startswith(FILE_PREFIX):
+                print(f"{key}: {value} --> not starting with '{FILE_PREFIX}', ignoring.")
+                continue
+        except AttributeError: # value.startswith
+            print(f"{key}: --> no a string, ignoring.")
+            continue
+
+        if not pathlib.Path(value).exists:
+            errors += 1
+
+            print(f"ERROR: {key}: {value} --> not found")
+            continue
+
+        # file found, nothing to do
+
+    return errors
+
+def traverse_role_vars():
+    errors = 0
+    for filename in glob.glob(ROLES_VARS_GLOB):
+        print()
+        print("###", filename)
+        with open(filename) as f:
+            try:
+                yaml_doc = yaml.safe_load(f)
+            except yaml.YAMLError as e:
+                print(f"--> invalid YAML file ({e})")
+                continue
+
+        if yaml_doc is None:
+            print(f"--> empty file")
+            continue
+
+        file_errors = validate_role_vars_files(filename, yaml_doc)
+        errors += file_errors
+
+        print(f"--> found {file_errors} error{'s' if file_errors > 1 else ''} in {filename}")
+
+    return errors
+
+if __name__ == "__main__":
+    print(f"INFO: Searching for missing files in '{ROLES_VARS_GLOB}'")
+    errors = traverse_role_vars()
```

Comment:
nit: It's best to avoid writing code directly under the `if __name...` conditional because variables defined here are in global scope, because they're not under any function.

Usually it's better to `def main():` and then put all this code inside it, then call `main()` from `if __name__ ==...`

Reply:
good point for the shadowing, I didn't think about that!